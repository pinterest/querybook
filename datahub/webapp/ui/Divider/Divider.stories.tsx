import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { Divider } from './Divider';

export default {
    title: 'Stateless/Divider',
    decorators: [centered],
};
export const _Divider = (args) => (
    <div style={{ width: '240px' }}>
        <div>Text Above</div>
        <Divider {...args} />
        <div>Text Below</div>
    </div>
);

_Divider.args = {
    marginTop: '8px',
    marginBottom: '4px',
    color: 'var(--color-accent)',
    height: '4px',
};
