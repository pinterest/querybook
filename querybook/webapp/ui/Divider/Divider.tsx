import styled from 'styled-components';

export const Divider = styled.hr<{
    marginTop?: string;
    marginBottom?: string;
    color?: string;
    height?: string;
}>`
    margin-top: ${(props) => (props.marginTop ? props.marginTop : '16px')};
    margin-bottom: ${(props) =>
        props.marginBottom ? props.marginBottom : '12px'};
    height: ${(props) => (props.height ? props.height : '1px')};
    background-color: ${(props) =>
        props.color ? props.color : 'var(--color-invert-3)'};
    margin-left: -4px;
    margin-right: -4px;
`;
