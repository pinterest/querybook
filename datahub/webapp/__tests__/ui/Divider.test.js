import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { Divider } from '../../ui/Divider/Divider';

it('renders without crashing', () => {
    shallow(<Divider />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<Divider />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(<Divider />);
        expect(output).toMatchSnapshot();
    });
});
