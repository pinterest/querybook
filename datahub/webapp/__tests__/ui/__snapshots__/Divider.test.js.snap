// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
    margin-top: ",
          [Function],
          ";
    margin-bottom: ",
          [Function],
          ";
    height: ",
          [Function],
          ";
    background-color: ",
          [Function],
          ";
    margin-left: -4px;
    margin-right: -4px;
",
        ],
      },
      "displayName": "styled.hr",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "hr",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<hr
  className="sc-bdVaJa fKCxyC"
/>
`;
